source/FreeRTOS_IP.c

```diff
@@ -1392,7 +1466,11 @@ BaseType_t xSendEventStructToIPTask( const IPStackEvent_t * pxEvent,
                      * IP task is already awake processing other message. */
                     vIPSetTCPTimerExpiredState( pdTRUE );
 
-                    if( uxQueueMessagesWaiting( xNetworkEventQueue ) != 0U )
+                    #if ipconfigEVENT_QUEUES > 1
+                        if( ulTaskGenericNotifyValueClear( xIPTaskHandle, 0, 0 ) != 0U )
```

Comment:
I believe `ulTaskNotifyValueClear` is the better version to use here to be consistent with all your other task notification API calls.